Orleans.Providers.MongoDB/StorageProviders/JsonBson/DefaultJsonBsonConverter.cs

```diff
@@ -83,35 +69,78 @@ public static BsonValue ToBson(this JToken source)
                 case JTokenType.TimeSpan:
                     return BsonValue.Create(((JValue)source).ToString());
                 case JTokenType.Date:
```

Comment:
I think you should introduce a virtual method for each case to make it easier to override it.

Reply:
Can do. That was my initial instinct, then after starting, I thought it was a bit over complicated. I will yield to your experience. Thank you.

Reply:
Here's a screenshot of partial changes using the concept described, if I understand correctly. I think this is what you mean ... I feel uncomfortable with it, how about you? 

![image](https://user-images.githubusercontent.com/1445862/46050478-ae0a9c00-c177-11e8-994f-20904d356b9b.png)

Reply:
I've pushed the complete set of changes.